src/core/tokenizer.ts

```diff
@@ -123,10 +123,20 @@ const tokenMatchersForLanguage: Partial<Record<LanguageId, Matcher>> =
   );
 
 export function getMatcher(languageId: string): Matcher {
-  return (
+  const matcher =
     tokenMatchersForLanguage[languageId as SupportedLanguageId] ??
-    defaultMatcher
-  );
+    defaultMatcher;
+
+  return {
+    ...matcher,
+    // TODO Why is this necessary???????????
+    identifierMatcher: new RegExp(matcher.identifierMatcher),
```

Comment:
I can't for the life of me understand why I need to create a new instance of this regex. As far as I can tell the old and new one are identical. Could we be running into a concurrency problem?

Reply:
@pokey Any ideas?

Reply:
What happens if you don't do that?

Reply:
It will fail to match in some of the test cases and we get no identifiers back causing the entire test to fail.